src/webgl/p5.RendererGL.js

```diff
@@ -631,7 +635,22 @@ p5.RendererGL.prototype._getColorShader = function () {
   return this._defaultColorShader;
 };
 
+p5.RendererGL.prototype.getTexture = function (img) {
+  // efficiency?
+  var f = function(element) {
+    return element.src === img;
+  };
+
+  var tex = this.textures.find(f);
+  if (tex === undefined) {
+    tex = new p5.Texture(this, img);
+    this.textures.push(tex);
+  }
 
+  tex.set(img);
```

Comment:
Got it. This was already happening in the code before; but a goal for this refactor was to make it easier to understand where an optimization might go. My plan is that `p5.Texture.set` can check the source image to see if any updates are needed, and then only do the texture upload if necessary.

The data itself may come from a p5.Image but can also be from a DOM video or img element (possibly containing an animated gif). I have the optimization working for p5.Image and think I can use the timestamp or check if the video is playing for the video, but I'm stumped on what to do for a DOM img with an animated gif. Any thoughts?